### PR TITLE
mimecast: add cloud integrated logs data stream

### LINKED ISSUE #11160: [Mimecast]Add support for Cloud Integrated Logs
https://developer.services.mimecast.com/siem-tutorial-ci
Logs are available using the [stream](https://developer.services.mimecast.com/docs/threatssecurityeventsanddataforci/1/routes/siem/v1/events/ci/get) and [batch](https://developer.services.mimecast.com/docs/threatssecurityeventsanddataforci/1/routes/siem/v1/batch/events/ci/get) apis.

- Entities (Message Summary) Logs
- Mail Flow logs
- Url Protect Logs(blocked events only)

Cloud Integrated is a new offering within the Mimecast platform. Only a subset of logs are available, so we'll have to clearly document which logs in our integration are available via Cloud Integrated (which Mimecast have agreed to assist with). 

Both Stream and Batch events are supported - Mimecast reccomend using the batch approach.

- [x] https://github.com/elastic/integrations/pull/11910

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

```
mimecast: add cloud integrated logs data stream

Tested against a real endpoint.

Pipeline test cases obtained from a test instance. Up to 10 examples of each
available type are included. Not all types are represented.
```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [ ] 

## Related issues

- Closes #11160
- Depends #11801

### CODE REVIEW

File: packages/mimecast/data_stream/cloud_integrated_logs/agent/stream/cel.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+            {
+              "events": body.value.map(b, has(b.url),
+                get(b.url).as(batch, batch.StatusCode == 200 ?
+                  bytes(batch.Body).mime("application/gzip").mime("application/x-ndjson").map(e,
+                    {
+                      "message": dyn(e.encode_json()),
+                    }
+                  )
+                :
+                    [{
```
Comment (efd6) [inline packages/mimecast/data_stream/cloud_integrated_logs/agent/stream/cel.yml.hbs:101]: Intentionally not logged.

File: packages/mimecast/data_stream/cloud_integrated_logs/_dev/test/pipeline/test-cloud-integrated-logs.log-expected.json [hunk 2]
```diff
@@ -0,0 +1,1143 @@
...
+            },
+            "event": {
+                "category": [
+                    "email"
+                ],
+                "created": "2024-11-27T16:20:25.453Z",
+                "original": "{\"_offset\":1825747,\"_partition\":53,\"accountId\":\"AUS2474\",\"aggregateId\":\"4Xz4RX59R6z84lm-ip3sazqogbn7c5haarz3ydiaeg_1732724425\",\"attachments\":[\"1-MB-Test_PCI-PII.xls\"],\"direction\":\"INBOUND\",\"historicalMail\":false,\"messageId\":\"<38989264-3456-4301-b2f9-22b8c78289db@SJ0PR18MB4090.namprd18.prod.outlook.com>\",\"policiesApplied\":[{\"action\":\"DO_NOTHING\",\"mode\":\"ACTIVE\",\"name\":\"Default O365 Mail policy\"}],\"processingId\":\"127b2e8d1334f481d9dda0fe16530c1de6aede980e5313957516c6adddefddbc_1732724425\",\"recipients\":[\"darron.giron@demovation-ci.b41.one\"],\"senderEnvelope\":\"\",\"senderHeader\":\"MicrosoftExchange329e71ec88ae4615bbc36ab6ce41109e@mc-ci.b41.one\",\"senderIp\":\"81.2.69.144\",\"source\":\"OFFICE_365_MAIL\",\"subject\":\"Undeliverable: Book Request\",\"subtype\":\"NO_DETECTIONS\",\"tags\":null,\"threatState\":\"DELIVERY_IN_PROGRESS\",\"threatType\":\"NO_DETECTIONS\",\"timestamp\":1732724425453,\"type\":\"entities\"}"
```
Comment (kcreddy) [inline packages/mimecast/data_stream/cloud_integrated_logs/_dev/test/pipeline/test-cloud-integrated-logs.log-expected.json:32]: There's a field named `recipients`. Can it be used for `email.to` and also into `related.user`?

**Review by kcreddy — APPROVED** [review]
nit. 
LGTM Otherwise 👍🏼

File: packages/mimecast/data_stream/cloud_integrated_logs/_dev/test/pipeline/test-cloud-integrated-logs.log-expected.json (hunk 2 above)
Comment (efd6) [reply packages/mimecast/data_stream/cloud_integrated_logs/_dev/test/pipeline/test-cloud-integrated-logs.log-expected.json:32]: That processor exists, but it was wrong.

**Pull request merged**